### ISSUE TITLE
404 error installing on Amazon Linux 2023

### ISSUE THREAD

**Issue by NathanBaulch** [openresty/openresty#927]
I get the following error whilst following the official [installation steps](https://openresty.org/en/linux-packages.html#amazon-linux) for Amazon Linux.

```shell
> sudo yum check-update
Official OpenResty Open Source Repository for Amazon Linux                                       5.1 kB/s |  13 kB     00:02
Errors during downloading metadata for repository 'openresty':
  - Status code: 404 for https://openresty.org/package/amazon/2023.1.20230906/x86_64/repodata/repomd.xml (IP: 54.183.167.213)
Error: Failed to download metadata for repo 'openresty': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

I'm new to OpenResty, Amazon Linux and yum, so apologies if it's something silly. I'm guessing the `/2023.1.20230906/` path segment is supposed to be simply `/2/` but I have no idea where yum took this value from.

**Comment by NathanBaulch** [comment]
OK, I switched to an older Amazon Linux 2 instance and it works, so perhaps this could be mentioned in the docs for n00bs like myself.

**Comment by zhuizhuhaomeng** [comment]
Thank you for your feedback, we will add it as soon as possible

**Comment by zhuizhuhaomeng** [comment]
We try to complete support for amazon Linux 2023 as soon as possible in Q1 2024.

**Comment by zhuizhuhaomeng** [comment]
Hi @NathanBaulch We have support Amazon 2023.
Feel free to use it.

**Comment by nickpearson** [comment]
Thank you @zhuizhuhaomeng for releasing an official build for Amazon Linux 2023. I tried installing it but am receiving the same 404 error as above, but with a newer version in the URL (`2023.3.20240122`):

```bash
# (whoami => root)
$ wget https://openresty.org/package/amazon/openresty.repo
$ mv openresty.repo /etc/yum.repos.d/
$ yum check-update
Official OpenResty Open Source Repository for Amazon Linux     3.0 kB/s |  13 kB     00:04    
Errors during downloading metadata for repository 'openresty':
  - Status code: 404 for https://openresty.org/package/amazon/2023.3.20240122/x86_64/repodata/repomd.xml (IP: 3.131.85.84)
Error: Failed to download metadata for repo 'openresty': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

Is it possible to get an OpenResty release for this version of Amazon Linux 2023? I'm not sure if it still just needs a symlink like @agentzh [commented](https://github.com/openresty/openresty/issues/346#issuecomment-365725530) on #346, but if so, hopefully that makes this a quick fix. Thank you!

cc @xiaocang (related to #947)

**Comment by nickpearson** [comment]
For anyone else having this same issue, I was able to work around it by editing the `openresty.repo` file and updating the `baseurl` value to use `latest` in place of `$releasever`. (This change essentially simulates the symlink @agentzh mentioned [here](https://github.com/openresty/openresty/issues/346#issuecomment-365725530). Instructions for getting that file are [here](https://openresty.org/en/linux-packages.html#amazon-linux))

Also worth noting: when running `sudo yum check-update`, the output shows that a newer version of Amazon Linux 2023 is available (2023.3.20240131), which looks like it was released just 9 days after the version I'm on (2023.3.20240122). If releases are this frequent, would it be possible for the project to provide an alternate repo file that uses `latest` (not `$releasever`) in its `baseurl`? Something like `openresty.latest.repo`?

If this is the right approach I can submit a PR for the `linux-packages` docs page and a PR to create the alternate `.repo` file (if you can point me to the GitHub repository for where that file should go).

**Comment by nickpearson** [comment]
Thank you @zhuizhuhaomeng. It appears you've fixed the issue for the two version for which you created a symbolic link. I just tried it:

```bash
$ wget https://openresty.org/package/amazon/openresty.repo
$ mv openresty.repo /etc/yum.repos.d/
$ sudo yum check-update
Amazon Linux 2023 repository                                  40 kB/s | 3.6 kB     00:00
Amazon Linux 2023 Kernel Livepatch repository                 27 kB/s | 2.9 kB     00:00
Official OpenResty Open Source Repository for Amazon Linux   7.7 kB/s |  18 kB     00:02
```

Just FYI, after the output shown above, `yum check-update` also shows `A newer release of "Amazon Linux" is available.` and lists two available version: 2023.3.20240131 and 2023.3.20240205. The second of these wasn't displayed when I ran the same command four days ago, so it appears releases do come quite frequently.

**Comment by centri-awilson** [comment]
Appreciate this is a few years old, but I've just had this occur to me. 

> Official OpenResty Open Source Repository for Amazon Linux                              6.3 kB/s |  29 kB     00:04
Errors during downloading metadata for repository 'openresty':
>  - Status code: 500 for https://openresty.org/package/amazon/2023.10.20260325/x86_64/repodata/7bd4a5735ac84cf48a4a8b642dbbd7d976883ce4a9e1464ae5e4f01bd66d93a8-primary.xml.gz (IP: 35.177.193.218)
>  - Status code: 500 for > https://openresty.org/package/amazon/2023.10.20260325/x86_64/repodata/bdff33fe0e8fc24c73f9d65d20d5cdc557f9639abe2cd7ed418bc4a0ebe3bec6-filelists.xml.gz (IP: 35.177.193.218)
Error: Failed to download metadata for repo 'openresty': Yum repo downloading error: Downloading error(s): repodata/7bd4a5735ac84cf48a4a8b642dbbd7d976883ce4a9e1464ae5e4f01bd66d93a8-primary.xml.gz - Cannot download, all mirrors were already tried without success; repodata/bdff33fe0e8fc24c73f9d65d20d5cdc557f9639abe2cd7ed418bc4a0ebe3bec6-filelists.xml.gz - Cannot download, all mirrors were already tried without success

`sed -i 's|amazon/\$releasever/|amazon/2023/|' /etc/yum.repos.d/openresty.repo`

~~To work around this.~~

That hasn't actually fixed it:

> [root@ip-10-0-1-250 ec2-user]# dnf install openresty
Last metadata expiration check: 0:00:35 ago on Tue Jul 14 17:09:39 2026.
Dependencies resolved.
========================================================================================================================
 Package                          Architecture         Version                            Repository               Size
========================================================================================================================
Installing:
 openresty                        x86_64               1.31.1.1-1.amzn2023                openresty               1.4 M
Installing dependencies:
 openresty-openssl3               x86_64               3.5.7-1.amzn2023                   openresty               2.5 M
 openresty-pcre2                  x86_64               10.47-1.amzn2023                   openresty               558 k
 openresty-zlib                   x86_64               1.3.2-1.amzn2023                   openresty                63 k
> 
> Transaction Summary
========================================================================================================================
Install  4 Packages
> 
> Total download size: 4.6 M
Installed size: 15 M
Is this ok [y/N]: Y
Downloading Packages:
[MIRROR] openresty-1.31.1.1-1.amzn2023.x86_64.rpm: Status code: 500 for https://openresty.org/package/amazon/2023/x86_64/openresty-1.31.1.1-1.amzn2023.x86_64.rpm (IP: 35.177.193.218)
[MIRROR] openresty-openssl3-3.5.7-1.amzn2023.x86_64.rpm: Status code: 500 for https://openresty.org/package/amazon/2023/x86_64/openresty-openssl3-3.5.7-1.amzn2023.x86_64.rpm (IP: 35.177.193.218)
[MIRROR] openresty-pcre2-10.47-1.amzn2023.x86_64.rpm: Status code: 500 for https://openresty.org/package/amazon/2023/x86_64/openresty-pcre2-10.47-1.amzn2023.x86_64.rpm (IP: 35.177.193.218)
[MIRROR] openresty-1.31.1.1-1.amzn2023.x86_64.rpm: Status code: 500 for https://openresty.org/package/amazon/2023/x86_64/openresty-1.31.1.1-1.amzn2023.x86_64.rpm (IP: 35.177.193.218)
[FAILED] openresty-1.31.1.1-1.amzn2023.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success
(2-3/4): openresty-pcre2-10.47-1.amzn202 31% [============-                           ] 516 kB/s | 1.4 MB     00:06 ETA
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Error downloading packages:
  openresty-1.31.1.1-1.amzn2023.x86_64: Cannot download, all mirrors were already tried without success
